solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -104,7 +104,7 @@
   private RestManager.Registry managedResourceRegistry;
 
   /** @see #reloadLuceneSPI() */
-  private boolean needToReloadLuceneSPI = false;
+  private boolean needToReloadLuceneSPI = false; // requires synchronization
```

Comment:
You mentioned in your reply that this needs to be `volatile`

Reply:
Yeah I did... but then I found it difficult to reason about the sequence of when it's set to true/false... vs when the actual work is done and so I instead made both methods that touch it synchronized to be clearer/safer.